apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -557,7 +557,9 @@ function handleCloseEditor( calypsoPort ) {
 			{
 				action: 'closeEditor',
 				payload: {
-					unsavedChanges: select( 'core/editor' ).isEditedPostDirty(),
+					unsavedChanges: editSitePackage
+						? select( 'core' ).__experimentalGetDirtyEntityRecords().length > 0
+						: select( 'core/editor' ).isEditedPostDirty(),
```

Comment:
Im not sure checking for `editSitePackage` makes sense here as we are using a core selector.  Also, `__experimentalGetDirtyEntityRecords` could make sense in the post editor as well if they are using reusable blocks or site blocks, etc.  (I just noticed this bug extends to the post editor if the parent post isnt dirty but other things like reusable blocks are).

Perhaps we should do something like:
`
unsavedChanges: select( 'core' ).__experimenalGetDirtyEntityRecords?.().length > 0 || select( 'core/editor' ).isEditedPostDirty()
`

Reply:
I was thinking about dropping `isEditedPostDirty` entirely. It feels safer to have them both for back compact.